p2panda-rs/src/operation/operation.rs

```diff
@@ -11,6 +11,55 @@ use crate::hash::Hash;
 use crate::operation::{OperationEncoded, OperationError, OperationFieldsError};
 use crate::Validate;
 
+/// Field type representing references to other documents.
+///
+/// The "relation" field type references a document id and the historical state which it had at the
+/// point this relation was created.
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+pub struct Relation {
+    /// Document id this relation is referring to.
+    document: Hash,
+
+    /// Reference to the exact version of the document.
+    ///
+    /// This field is `None` when there is no more than one operation (when the document only
+    /// consists of one CREATE operation).
```

Comment:
This could be something we need to define on a protocol level: Do we want to give users the option to omit this field in UPDATE operations?

Allowing it means that users can decide themselves if they want to support immutable data or not. Disallowing it means we force developers to take care of it (they can still fill it with garbage of course).

Reply:
What would you prefer?

Reply:
I prefer leave it optional. If it isn't included, it's assumed the exact version doesn't matter, just grab the latest. This will also probably be the usual behaviour, it's just some cases may want it to be more specific. Although, if it _is_ required for some application, then I don't know how that would be shown.... 

We can specify more clearly at system schema level in any case if we want to, so it's not a problem there.